File: services/products/src/main/java/br/com/deltaglobalbank/products/domain/product/ProductType.java
package br.com.deltaglobalbank.products.domain.product;

import java.util.Arrays;

public enum ProductType {
    LENDING;

    public String toDatabaseValue() {
        return name();
    }

    public static ProductType fromDatabaseValue(String value) {
        return Arrays.stream(values())
            .filter(type -> type.name().equalsIgnoreCase(value))
            .findFirst()
            .orElseThrow(() -> new IllegalArgumentException("invalid_product_type: " + value));
    }
}

File: services/products/src/main/kotlin/br/com/deltaglobalbank/products/domain/product/ProductType.kt
package br.com.deltaglobalbank.products.domain.product

enum class ProductType {
    LENDING;
    fun toDatabaseValue(): String = name
    companion object {
        fun fromDatabaseValue(value: String): ProductType =
            entries.firstOrNull { it.name.equals(value, ignoreCase = true) }
                ?: throw IllegalArgumentException("invalid_product_type: $value")
    }
}